### ISSUE TITLE
Incorrect prompt width when control chars are used

### ISSUE THREAD

**Issue by gwenn** [kkawakam/rustyline#22]
```
extern crate unicode_width;

fn main () {
    let s = "\x1b[1;32m>>\x1b[0m ";
    println!("{}", unicode_width::UnicodeWidthStr::width(s)); // 12
    let s = ">> ";
    println!("{}", unicode_width::UnicodeWidthStr::width(s)); // should be 3
}
```

**Comment by gwenn** [comment]
I tried `term_grid` crate with no success:
```
extern crate term_grid;

use term_grid::Cell;

fn main () {
    let s = "\x1b[1;32m>>\x1b[0m ";
    let cell: Cell = s.into();
    println!("{}", cell.width); // 12
}
```

**Comment by kkawakam** [comment]
Fixed by PR #23
